File: backend/app/schemas/boveda.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class CredencialCreate(BaseModel):
    id_empresa: int
    nombre: str
    usuario_acceso: Optional[str] = None
    password_plain: str  # Will be encrypted before storing
    url: Optional[str] = None
    notas: Optional[str] = None


class CredencialUpdate(BaseModel):
    nombre: Optional[str] = None
    usuario_acceso: Optional[str] = None
    password_plain: Optional[str] = None
    url: Optional[str] = None
    notas: Optional[str] = None


class CredencialOut(BaseModel):
    id_credencial: int
    id_empresa: int
    nombre: str
    usuario_acceso: Optional[str] = None
    url: Optional[str] = None
    notas: Optional[str] = None
    # password_plain only returned when explicitly requested and MFA validated

    model_config = {"from_attributes": True}


class CredencialWithPassword(CredencialOut):
    password_plain: str

File: frontend/src/types/boveda.ts
export interface Credencial {
  id_credencial: number
  id_empresa: number
  nombre: string
  usuario_acceso?: string
  url?: string
  notas?: string
}

export interface CredencialReveal extends Credencial {
  password_plain: string
}

export interface CredencialCreatePayload {
  id_empresa: number
  nombre: string
  usuario_acceso?: string
  password_plain: string
  url?: string
  notas?: string
}

export interface CredencialUpdatePayload {
  nombre?: string
  usuario_acceso?: string
  password_plain?: string
  url?: string
  notas?: string
}
